web_server_v1 IDF fails to compile

### The problem

Compiling for an ESP32-C6 with the V1 webserver will fail due to a missing stream write function. 

### Which version of ESPHome has the issue?

2026.1.0-dev

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP32

### Component causing the issue

web_server

### YAML Config

```yaml
esphome:
  name: idftest
  friendly_name: IDFTest

esp32:
  board: esp32-c6-devkitc-1
  flash_size: 4MB
  variant: esp32c6
  framework:
    type: esp-idf
    version: recommended

wifi:
  ssid: bla

ota:
  - platform: web_server

web_server:
  port: 80
  version: 1
```

### Anything in the logs that might be useful for us?

```txt
Compiling .pioenvs/grpm2-324e18/src/esphome/components/web_server_idf/utils.cpp.o
src/esphome/components/web_server/web_server_v1.cpp: In function 'void esphome::web_server::write_html_escaped(esphome::web_server_idf::AsyncResponseStream*, const char*)':
src/esphome/components/web_server/web_server_v1.cpp:25:17: error: 'class esphome::web_server_idf::AsyncResponseStream' has no member named 'write'
   25 |         stream->write(*p);
      |                 ^~~~~
Compiling .pioenvs/grpm2-324e18/src/esphome/components/web_server_idf/web_server_idf.cpp.o
*** [.pioenvs/grpm2-324e18/src/esphome/components/web_server/web_server_v1.cpp.o] Error 1
```

### Additional information

_No response_

--- a/esphome/components/web_server_idf/web_server_idf.h
+++ b/esphome/components/web_server_idf/web_server_idf.h
@@ -80,6 +80,7 @@ class AsyncResponseStream : public AsyncWebServerResponse {
   void print(const std::string &str) { this->content_.append(str); }
   void print(float value);
   void printf(const char *fmt, ...) __attribute__((format(printf, 2, 3)));
+  void write(uint8_t c) { this->content_.push_back(static_cast<char>(c)); }
 
  protected:
   std::string content_;

--- a/tests/components/web_server/test_v1.esp32-idf.yaml
+++ b/tests/components/web_server/test_v1.esp32-idf.yaml
@@ -1,0 +1,1 @@
+<<: !include common_v1.yaml

